netket/operator/_hamiltonian.py

```diff
@@ -219,7 +216,9 @@ def copy(self, *, dtype: Optional[DType] = None):
             dtype = self.dtype
 
         graph = Graph(edges=[list(edge) for edge in self.edges])
-        return Ising(hilbert=self.hilbert, graph=graph, J=self.J, h=self.h, dtype=dtype)
+        return type(self)(
```

Comment:
this is a door open to bugs...
please just define it on the final concrete class.